### PR TITLE
Show warnings if package versions do not match

### PR DESCRIPTION
**Pull request by frankenjoe**
API:

image

image

Tutorial:

image

### CODE REVIEW

File: audobject/core/config.py [hunk 1]
```diff
@@ -3,6 +3,10 @@
 
 class config:
 
+    PACKAGE_MISMATCH_WARN_LEVEL = define.PackageMismatchWarnLevel.STANDARD
```
Comment (hagenw) [inline audobject/core/config.py:6]: At the moment the configuration entry looks a little bit empty:

image

compared e.g. with https://audeering.github.io/audb/api.html#config

So maybe we add a docstring as well:

```suggestion
    r"""Get/set configuration values for the :mod:`audobject` module.
    
    Example:
        
        >>> config.PACKAGE_MISMATCH_WARN_LEVEL
        1
        >>> config.PACKAGE_MISMATCH_WARN_LEVEL = 0
        >>> config.PACKAGE_MISMATCH_WARN_LEVEL
        0
        
    """
    
    PACKAGE_MISMATCH_WARN_LEVEL = define.PackageMismatchWarnLevel.STANDARD
```

File: docs/usage.rst [hunk 2]
```diff
@@ -858,6 +858,29 @@ And convert it to YAML.
...
+
+    __version__ = '1.1.0'
+
+    o2 = audobject.from_yaml_s(o_yaml)
+    print(o2)
+
+But if we load it with an older version,
+a warning will be shown.
+When a warning should be shown can be controlled with
+:attr:`audobject.config.PACKAGE_MISMATCH_WARN_LEVEL`.
```
Comment (hagenw) [inline docs/usage.rst:874]: I would be a little bit more explicit here, maybe:

```suggestion
We can force it to show no warning at all,
or always show a warning
when the package version does not match
by adjusting
:attr:`audobject.config.PACKAGE_MISMATCH_WARN_LEVEL`.
```

File: tests/test_warnings.py [hunk 3]
```diff
@@ -79,3 +83,98 @@ def __init__(
...
+    $audobject.core.testing.TestObject:
+      name: test
+    '''
+
+    audobject.config.PACKAGE_MISMATCH_WARN_LEVEL = level
+
+    with warnings.catch_warnings():
+        audobject.from_yaml_s(yaml_s)
+
+for level, version, expected in [
```
Comment (hagenw) [inline tests/test_warnings.py:112]: Is it easier this way then using `@pytest.mark.parametrize`?

Comment (frankenjoe) [reply tests/test_warnings.py]: No of course not, changed it.

Comment (hagenw) [reply tests/test_warnings.py]: :)

**Pull request merged**